rfc/0014-attribute-based-valuetypes/README.md

```diff
@@ -42,47 +37,61 @@ valuetype CorrelationCoefficient oftype decimal {
 }
 ```
 
-This syntax is smart in that you don't have to list and name an attribute but rather rely on an implicit 'value' attribute. Still, I propose to make that attribute explicit. New syntax would be:
+This syntax is smart in that you don't have to list and name an attribute but rather rely on an implicit 'value' attribute.
+
+The current syntax, in which an underlying value type is referenced through 'oftype', it does not use instantiation/inheritance but rather composition.
+The purpose of using composition is that it is
+
+1. more similar to traditional approaches and
+2. is needed anyway for multi-attribute value types.
+
+Thus, this RFC removes the `oftype` keyword in a value type definitions and proposes an attribute-based syntax that is extendable to a future multi-attribute syntax.
+
+## Explanation
+
+I propose to make that attribute explicit. The new syntax would be:
 
 ```jayvee
 valuetype CorrelationCoefficient {
-  attributes: [
-    value oftype decimal;
-  ];
-  constraints: [
-    MinusOneToPlusOneRange; // Don't know how to attach this to value attribute
-  ];
+  property correlation oftype decimal;
+
+  constraint minusOneToPlusOneRange: MinusOneToPlusOneRange(value = correlation);
 }
+
+constraint MinusOneToPlusOneRange on decimal:
+  value >= -1 and value <= 1;
 ```
 
-While more verbose, it prepares the way for 
+The syntax is kept flat rather than using arrays.
+
+Despite referencing reusable constraints, the syntax also allows in-place constraint definition:
 
 ```jayvee
-valuetype Money {
-  attributes: [
-    amount oftype decimal;
-    currency oftype Currency;
-  ];
+valuetype CorrelationCoefficient {
+  property correlation oftype decimal;
+
+  constraint minusOneToPlusOneRange:
+    correlation >= 1 and correlation <=1;
 }
 ```
-which we'll need anyway. Conceivably, the step to multi-attribute value types could be merged with this one, but I simply wanted to try the RFC process rather than keep sending email ;-)
-
-<!-- 
-  TODO: Explain the details of the RFC. 
-  If the RFC contains more than a single cohesive aspect, structure this section accordingly.
-  Make sure to provide realistic modelling examples on the example data set introduced above.
--->
 
 ## Drawbacks
 
-1. Introduces a redundant syntax,
-2. Creates extra work/may be too difficult, and
-3. May be disruptive to how the language currently works. 
+1. No backward compatible syntax (breaking change).
+2. Removes the inheritance mechanism.
 
 ## Alternatives
 
-Conceivably, we could use multiple inheritance for multi-attribute value types... just joking. 
+1. Keep the current syntax.
+2. Use arrays instead of the flat syntax.
```

Comment:
Remember to remove this if we apply @rhazn's [suggestion](https://github.com/jvalue/jayvee/pull/595/files#r1650751505)